contracts/modules/TransferManager/ManualApprovalTransferManager.sol

```diff
@@ -19,34 +19,34 @@ contract ManualApprovalTransferManager is ITransferManager {
 
     //Manual approval is an allowance (that has been approved) with an expiry time
     struct ManualApproval {
+        address from;
+        address to;
         uint256 allowance;
         uint256 expiryTime;
+        bytes32 description;
     }
 
-    //Manual blocking allows you to specify a list of blocked address pairs with an associated expiry time for the block
-    struct ManualBlocking {
-        uint256 expiryTime;
-    }
-
-    //Store mappings of address => address with ManualApprovals
-    mapping (address => mapping (address => ManualApproval)) public manualApprovals;
-
-    //Store mappings of address => address with ManualBlockings
-    mapping (address => mapping (address => ManualBlocking)) public manualBlockings;
+    mapping (address => mapping (address => uint256)) public approvalIndex;
+    mapping (address => mapping (address => bool)) public hasApproval;
```

Comment:
Rather than maintaining `hasApproval`, we can just set approvalIndex as index + 1 and then if approvalIndex = 0, that means approval is not present. 

Setting hasApproval will take ~40k gas and it can add up to a lot when using multi functions.